cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -29,10 +29,20 @@
 #include "arrow/filesystem/localfs.h"
 #include "arrow/filesystem/path_util.h"
 #include "arrow/filesystem/util_internal.h"
+#include "arrow/util/checked_cast.h"
+#include "arrow/util/make_unique.h"
 
 namespace arrow {
+
+using internal::checked_cast;
+using internal::make_unique;
+
 namespace engine {
 
+using SubstraitConverter = std::function<Result<std::unique_ptr<substrait::Rel>>(
+    const std::shared_ptr<Schema>&, const compute::Declaration&, ExtensionSet*,
+    const ConversionOptions&)>;
+
```

Comment:
I think you can get rid of this now that we aren't doing a registry approach?

Reply:
I added it because the parameter would be too long. Just for the readability and usage.

Reply:
Yes we used it before with the `registry` but kept it for the above mentioned reason.

Reply:
This is no longer needed. The refactor removes the generic interface and replaces with specific types.